app/models/course/condition/achievement.rb

```diff
@@ -67,4 +71,9 @@ def validate_unique_dependency
     return unless required_achievements_for(conditional).include?(achievement)
     errors.add(:achievement, :unique_dependency)
   end
+
+  # Trigger for resolving the conditional for a course user
+  Course::UserAchievement.after_save do |achievement|
+    execute_after_commit { Course::Condition::Achievement.on_dependent_status_change(achievement) }
```

Comment:
Does that means we are adding the instance method to the Course::UserAchievement class? Since the after_save will be done within the scope of the UserAchievement.